core/src/io/anuke/mindustry/world/blocks/power/PowerDiode.java

```diff
@@ -0,0 +1,85 @@
+package io.anuke.mindustry.world.blocks.power;
+
+import io.anuke.arc.Core;
+import io.anuke.arc.graphics.g2d.Draw;
+import io.anuke.arc.graphics.g2d.TextureRegion;
+import io.anuke.arc.math.Mathf;
+import io.anuke.mindustry.world.Block;
+import io.anuke.mindustry.world.Tile;
+
+import static io.anuke.mindustry.Vars.world;
+
+public class PowerDiode extends Block{
+
+    protected TextureRegion arrow;
+
+    public PowerDiode(String name){
+        super(name);
+        rotate = true;
+        update = true;
+        solid = true;
+        insulated = true;
+    }
+
+    @Override
+    public void update(Tile tile){
+        super.update(tile);
+
+        Tile back = getNearby(tile, (tile.rotation() + 2) % 4);
+        Tile front = getNearby(tile, tile.rotation());
+
+        if(!back.block().hasPower || !front.block().hasPower) return;
+        PowerGraph backGraph = back.entity.power.graph;
+        PowerGraph frontGraph = front.entity.power.graph;
+        if(backGraph == frontGraph) return;
+
+        // skip if the receiving graph is already full
+        if(frontGraph.getBatteryStored() == frontGraph.getTotalBatteryCapacity()) return;
+
+        // half the difference
+        float send = Mathf.clamp((backGraph.getBatteryStored() - frontGraph.getBatteryStored()) / 2, 0f, Integer.MAX_VALUE);
+
+        // send all overflow if all batteries of the sending graph are full
+        if(backGraph.getBatteryStored() == backGraph.getTotalBatteryCapacity()){
+            send += backGraph.getLastPowerProduced() - backGraph.getPowerNeeded();
+        }
+
+        // limit offering to space available
+        send = Mathf.clamp(send, 0f, frontGraph.getTotalBatteryCapacity() - frontGraph.getBatteryStored());
+
+        // limit to sendable power
+        send = Mathf.clamp(send, 0f, backGraph.getBatteryStored());
+
+        if (send == 0f) return;
+        backGraph.useBatteries(send);
+        frontGraph.chargeBatteries(send);
+    }
+
+    @Override
+    public void load(){
+        super.load();
+        arrow = Core.atlas.find(name + "-arrow");
+    }
+
+    @Override
+    public void draw(Tile tile){
+        Draw.rect(region, tile.drawx(), tile.drawy(), 0);
+        Draw.rect(arrow, tile.drawx(), tile.drawy(), rotate ? tile.rotation() * 90 : 0);
```

Comment:
You should also override the `drawRequestRegion(...)` method to also rotate it properly, since the block looks weird in schematics/placement (as the whole thing is rotated there).

Reply:
i did try that as well originally, but my ide really didn't like that at all 🙄 

![Screen Shot 2019-11-04 at 14 29 32](https://user-images.githubusercontent.com/3179271/68124415-a76c5d80-ff0f-11e9-8186-5b7c05ca47a3.png)
![Screen Shot 2019-11-04 at 14 29 37](https://user-images.githubusercontent.com/3179271/68124416-a76c5d80-ff0f-11e9-8703-79ac726a7ef4.png)
![Screen Shot 2019-11-04 at 14 29 41](https://user-images.githubusercontent.com/3179271/68124417-a76c5d80-ff0f-11e9-913a-3975fcf23979.png)
<img width="725" alt="Screen Shot 2019-11-04 at 14 28 46" src="https://user-images.githubusercontent.com/3179271/68124418-a76c5d80-ff0f-11e9-9433-2fd9252a1648.png">
<img width="915" alt="Screen Shot 2019-11-04 at 14 29 08" src="https://user-images.githubusercontent.com/3179271/68124419-a804f400-ff0f-11e9-8581-74b8fa1980ea.png">

Reply:
Refresh your dependencies or remove/update your local version of Arc. That interface was moved about a week ago.

Reply:
i did use the gradle refresh flag, without result.

Reply:
tried --refresh-dependencies & removing the gradle cache folder, issues persist 🤔

Reply:
apparently also had to knock some sence into intelij 😞